playwright/_impl/_json_pipe.py

```diff
@@ -48,18 +46,15 @@ async def wait_until_stopped(self) -> None:
     async def connect(self) -> None:
         self._stopped_future: asyncio.Future = asyncio.Future()
 
-        close_error: Optional[str] = None
+        def handle_closed(close_error: str = None) -> None:
+            self.emit("close", close_error)
+            self._stopped_future.set_result(None)
 
         def handle_message(message: Dict) -> None:
             try:
                 self.on_message(cast(ParsedMessagePayload, message))
             except Exception as e:
-                nonlocal close_error
-                close_error = str(e)
-
-        def handle_closed() -> None:
-            self.emit("close", close_error)
-            self._stopped_future.set_result(None)
+                handle_closed(str(e))
```

Comment:
I wonder if you want to close the pipe instead?

Reply:
we can just remove this code if you want, feel like its not covered by test anyways.